### PR TITLE
Inheritance

### LINKED ISSUE #109: Look into inheritance between GameTeam and Games
- How can we utilize inheritance to allow for GameTeam to populate season_id based on Game?

### PR DESCRIPTION
**Pull request by AngelaGuardia**
**Added**
- Created a new DataSet superclass that will be inherited by Team, Game, and GameTeams

**Changed**
- Moved repeated methods from each subclass into DataSet (create, count)
- Added an "all" class method to each subclass
- Added tests to each subclass for the new inherited supermethods

closes #109

### CODE REVIEW

File: lib/data_set.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    all.count
+  end
+
+  def self.find(attribute, value)
+    all.find { |element| element.send(attribute) == value }
+  end
+
+  def self.find_all(attribute, value)
+    all.find_all { |element| element.send(attribute) == value }
+  end
```
Comment (AngelaGuardia) [inline lib/data_set.rb:28]: We may not have time to refactor our current StatTracker methods to use these instead. I'm okay with that!

**Review by hopesgit — APPROVED** [review]

File: lib/data_set.rb (hunk 1 above)
Comment (leahriffell) [reply lib/data_set.rb:28]: Good use of the send method here!

**Review by leahriffell — APPROVED** [review]
Looks good to me. Thank you!

File: lib/data_set.rb (hunk 1 above)
Comment (sarynm12) [reply lib/data_set.rb:28]: Nice! That looks good!

**Review by sarynm12 — APPROVED** [review]

**Pull request merged**